nixos/tests/virtualbox.nix

```diff
@@ -29,6 +45,10 @@ let
       "${pkgs.dbus.daemon}/bin/dbus-daemon" --fork \
         --config-file="${pkgs.dbus.daemon}/share/dbus-1/system.conf"
 
+      # Some programs (e.g. VBoxService, systemd-detect-virt) load libdbus
+      # using dlopen(), make sure that works.
+      export LD_LIBRARY_PATH=${pkgs.dbus.lib}/lib
```

Comment:
I figured that when I refactored the additions package to not skip the fixup phase which includes RUNPATH stripping, the dbus path got stripped out since there was no direct dependency on dbus. I will fix this in a new commit by adding the dbus path in `postFixup` (and removing dbus from the first round of patchelf because none of the binaries directly depend on dbus). This leaves `VBoxClient` and `VBoxService` with dbus in RUNPATH (as found by grepping binaries for `libdbus-1.so`). So I think that explains why `VBoxService` broke in the initrd.

As far as `systemd-detect-virt` is concerned, I suspect what happened is that because `mount.vboxsf` previously ended up with a bunch of unnecessary `RUNPATH` entries, `copy_bin_and_libs "${guestAdditions}/bin/mount.vboxsf"` resulted in libdbus being copied to /lib (or something equivalent, not sure what exactly it does) and found by `systemd-detect-virt`. What's interesting is that just running `systemd-detect-virt` in NixOS does not result in a libdbus error and strace shows that it does not load libdbus (but I see no point in investigating why).

So I will remove the `LD_LIBRARY_PATH` setting in the tests from its current place and add it just for the `systemd-detect-virt` call. It would not work without this now because because `mount.vboxsf` no longer has dbus in RUNPATH (it doesn't need it) and we don't `copy_bin_and_libs` anything that does.